pkg/inject/inject.go

```diff
@@ -539,6 +546,32 @@ func (conf *ResourceConfig) serviceAccountVolumeMount() *corev1.VolumeMount {
 	return nil
 }
 
+func (conf *ResourceConfig) trace() *charts.Trace {
+	var (
+		svcAddr    = conf.getOverride(k8s.ProxyTraceCollectorSvcAddr)
+		svcAccount = conf.getOverride(k8s.ProxyTraceCollectorSvcAccount)
```

Comment:
Shouldn't this be defaulted to `default` if the annotation is missing?

Reply:
The default is read from `values.yaml`, like all other defaults.

Reply:
I think you're missing a step. `conf.getOverride()` only checks for annotations. It seems to me you need to add the `Trace` struct into the protobuf def, and then if `conf.getOverride()` doesn't return anything, default to something like `conf.configs.GetProxy().GetTrace().GetCollectorSvcAddr()` and `conf.configs.GetProxy().GetTrace().GetCollectorSvcAccount()`

Reply:
The reason why I didn't update the protobuf is because everything in the pb def (i.e. all the configuration in the `linkerd-config` config map) is meant to apply to *all* proxies, and be persisted across upgrades. Since #3472 is limited to (on demand) enabling tracing per workload, I think we don't need to add these to the pb def.

In this case, if `conf.getOverride()` returns an empty string, implying that no tracing annotations are added, we assume tracing is disabled.

Reply:
Ok, in that case what are the `Trace` properties in `values.yaml` for, since we only care for the annotation values?

Reply:
Ha.. you are right. It was a leftover from pre- a4b17184add0b3f6b9579a459e32e9ad3aea1bdc.

Reply:
Ha.. you are right. It was a leftover from pre- a4b17184add0b3f6b9579a459e32e9ad3aea1bdc.

Reply:
Per slack chat, we will keep these as they are needed for https://github.com/linkerd/linkerd2/issues/3468.